src/ir/module-utils.h

```diff
@@ -444,10 +461,57 @@ collectSignatures(Module& wasm,
       counts[innerPair.first] += innerPair.second;
     }
   }
+  // We must sort all the dependencies of a signature before it. For example,
```

Comment:
I don't think this is actually true in the proposal, but I opened https://github.com/WebAssembly/function-references/issues/41 to clarify that. I do recognize that this is required by the current parser in Binaryen, though, because I haven't done the work to allow using undefined types in type definitions yet. I will hop on that.

Reply:
Resolved offline.